plasma_framework/python_tests/tests/contracts/root_chain/test_process_exits.py

```diff
@@ -1002,3 +1002,49 @@ def test_challenged_standard_exit_does_not_block_ife_output_exit(testlang, plasm
     # and does not get the tokens back
     alice_token_balance = token.balanceOf(alice.address)
     assert alice_token_balance == alice_token_balance_before
+
+
+def test_after_canonical_ife_is_finalized_inputs_are_not_exited_when_ife_is_restarted_and_non_canonical(testlang, plasma_framework, token):
```

Comment:
Though I am generally not a fan of adding too many comments, but I think it might worth to write down the Alice Bob description for each test. It has become quite complex and the function name honestly would probably not possible to be clear enough if first seeing this without any background info.